server/chat.ts

```diff
@@ -1852,25 +1852,14 @@ export const Chat = new class {
 				this.loadPluginDirectory(path, depth);
 			} else {
 				try {
-					this.loadPlugin(path);
+					this.loadPluginData(require(path), path.split('/').pop()?.slice(0, -3) || path);
```

Comment:
`path.split('/').pop()?.slice(0, -3)` is wrong for TSX files, isn't it?

Reply:
Also I think we should go back to using `loadPlugin`. The entire point is to not load files that aren't `.ts` or `.tsx` or `.js` or `.jsx` files. For instance, if there's a `Thumbs.db` or `.DS_Store` file.